tests/src/drivers/cipher.c

```diff
@@ -0,0 +1,348 @@
+    if( output_size < test_driver_cipher_forced_output_length )
+        return PSA_ERROR_BUFFER_TOO_SMALL;
+
+    memcpy(output, test_driver_cipher_forced_output, test_driver_cipher_forced_output_length);
+    *output_length = test_driver_cipher_forced_output_length;
+
+    return test_transparent_cipher_status;
+}
+
+psa_status_t test_transparent_cipher_decrypt(
+    const psa_key_attributes_t *attributes,
+    const uint8_t *key, size_t key_length,
+    psa_algorithm_t alg,
+    const uint8_t *input, size_t input_length,
+    uint8_t *output, size_t output_size, size_t *output_length)
+{
+    (void) attributes;
+    (void) key;
+    (void) key_length;
+    (void) alg;
+    (void) input;
+    (void) input_length;
+    test_transparent_cipher_hit++;
+
+    if( test_transparent_cipher_status != PSA_SUCCESS )
+        return test_transparent_cipher_status;
+    if( output_size < test_driver_cipher_forced_output_length )
+        return PSA_ERROR_BUFFER_TOO_SMALL;
+
+    memcpy(output, test_driver_cipher_forced_output, test_driver_cipher_forced_output_length);
+    *output_length = test_driver_cipher_forced_output_length;
+
+    return test_transparent_cipher_status;
+}
+
+psa_status_t test_transparent_cipher_encrypt_setup(
+    test_transparent_cipher_operation_t *operation,
+    const psa_key_attributes_t *attributes,
+    const uint8_t *key, size_t key_length,
+    psa_algorithm_t alg)
+{
+    (void) attributes;
+    (void) key;
+    (void) key_length;
+    (void) alg;
+
+    /* write our struct, this will trigger memory corruption failures
+     * in test when we go outside of bounds. */
+    memset(operation, 0, sizeof(test_transparent_cipher_operation_t));
+
+    test_transparent_cipher_hit++;
+    return test_transparent_cipher_status;
+}
+
+psa_status_t test_transparent_cipher_decrypt_setup(
+    test_transparent_cipher_operation_t *operation,
+    const psa_key_attributes_t *attributes,
+    const uint8_t *key, size_t key_length,
+    psa_algorithm_t alg)
+{
+    (void) attributes;
+    (void) key;
+    (void) key_length;
+    (void) alg;
+
+    /* write our struct, this will trigger memory corruption failures
+     * in test when we go outside of bounds. */
+    memset(operation, 0, sizeof(test_transparent_cipher_operation_t));
+
+    test_transparent_cipher_hit++;
+    return test_transparent_cipher_status;
+}
+
+psa_status_t test_transparent_cipher_abort(
+    test_transparent_cipher_operation_t *operation)
+{
+    /* write our struct, this will trigger memory corruption failures
+     * in test when we go outside of bounds. */
+    memset(operation, 0, sizeof(test_transparent_cipher_operation_t));
+
+    test_transparent_cipher_hit++;
+    return test_transparent_cipher_status;
+}
+
+psa_status_t test_transparent_cipher_generate_iv(
+    test_transparent_cipher_operation_t *operation,
+    uint8_t *iv,
+    size_t iv_size,
+    size_t *iv_length)
+{
+    (void) operation;
+    (void) iv;
+    (void) iv_size;
+    (void) iv_length;
+
```

Comment:
Ditto, please add the "usual" code for negative testing.

Reply:
done